tsl/src/chunk_api.c

```diff
@@ -485,8 +483,8 @@ chunk_api_create_on_data_nodes(Chunk *chunk, Hypertable *ht)
 			DatumGetCString(values[AttrNumberGetAttrOffset(Anum_create_chunk_schema_name)]);
 		table_name = DatumGetCString(values[AttrNumberGetAttrOffset(Anum_create_chunk_table_name)]);
 
-		if (namestrcmp(&chunk->fd.schema_name, schema_name) != 0 ||
-			namestrcmp(&chunk->fd.table_name, table_name) != 0)
+		if (namestrcmp((Name) &chunk->fd.schema_name, schema_name) != 0 ||
+			namestrcmp((Name) &chunk->fd.table_name, table_name) != 0)
```

Comment:
Not sure why these casts are here. Neither the const-ness of `schema_name` nor the `table_name` has changed, so why are these necessary?

Reply:
Check the signature of `namestrcmp`. For some reason, `namestrcmp` takes a mutable (non-const) `Name` and a `const` string (`const char *`) and compares them. Normally, `strcmp` takes two `const` pointers, but not in PostgreSQL apparently. I cast it since I know that `namestrcmp` does not mutate the `Name` argument.

Reply:
The type of `chunk->fd.schema_name` is `NameData` and AFAICT you haven't changed that in this commit. Taking the address of `chunk->fd.schema_name` gives a `NameData*`, which is the same as `Name`. This is what `namestrcmp` accepts as the first argument, so that would mean that cast is redundant. What am I missing?

Reply:
the chunk is `const Chunk *`, which means that also `chunk->fd.schema_name` is `const NameData *`. The function `namestrcmp` doesn't accept this type since it expects a non-const object.